### ISSUE TITLE
Alizer SelectDevFileFromTypes matches PHP project as nodejs project

### ISSUE THREAD

**Issue by yangcao77** [devfile/api#1219]
## Which area this feature is related to?

/kind bug

### Which area this bug is related to?

/area alizer

### Go project

**Operating System and version:**

**Go Pkg Version:**

### Node.js project

**Operating System and version:**

**Node.js version:**

**Yarn version:**

**Project.json:**

### Web browser

**Operating System and version:**

**Browser name and version:**

## Bug Summary

**Describe the bug:**

With latest Alizer, please run this main.go file, this simplifies how HAS runs Alizer
[main.txt](https://github.com/devfile/api/files/12304938/main.txt)

The result:
```
alizerDevfileTypes is [{nodejs-basic JavaScript Node.js [Node.js Express] []} {code-with-quarkus Java Quarkus [Java Quarkus] []} {java-springboot-basic Java springboot [Java Spring] []} {python-basic Python Python [Python Pip Flask] []} {go-basic Go Go [Go] []} {dotnet-basic .NET dotnet [.NET] []}] 
lenth of results 1 
**************
component devfile-stack-php-laravel,path: /Users/stephanie/Downloads/devfile-stack-php-laravel/ 
 languages: [{PHP [inc] 100 [Laravel] [] true false}] 
**************
detectedType: {nodejs-basic JavaScript Node.js [Node.js Express] []} 
detectedDevfileEndpoint: https://registry.devfile.io/devfiles/nodejs-basic 
```

With previous version of Alizer:
```
alizerDevfileTypes is [{nodejs-basic JavaScript Node.js [Node.js Express]} {code-with-quarkus Java Quarkus [Java Quarkus]} {java-springboot-basic Java springboot [Java Spring]} {python-basic Python Python [Python Pip Flask]} {go-basic Go Go [Go]} {dotnet-basic .NET dotnet [.NET]}] 
lenth of results 0 
```

Seems like latest Alizer detects the PHP component, but cannot filter out when calling `SelectDevFileFromTypes`

**To Reproduce:**

## Expected behavior

Alizer can detect PHP as a component, but should not match it to any devfile.

**Comment by thepetk** [comment]
@yangcao77 inside the `main.txt` I see there is a line inside `getAlizerDevfileTypes`:
```golang
registryIndex, err := registryLibrary.GetRegistryIndex(registryURL, registryLibrary.RegistryOptions{
		Telemetry: registryLibrary.TelemetryData{},
	}, schema.SampleDevfileType)
```
If we update to `schema.StackDevfileType` I'm able to solve this issue. As a result, it doesn't seem to me that it is an issue inside alizer's code. WDYT?

**Comment by yangcao77** [comment]
@thepetk For HAS, we want to use devfiles from samples, so `schema.SampleDevfileType` is exactly what we want.

**Comment by thepetk** [comment]
@yangcao77 I've created the [PR](https://github.com/devfile/alizer/pull/22). In case of no devfile found for components it applies language analysis only for the main language of the given source code. The result of the `main.txt` with this workaround is:

```
** Checking recognizer.SelectDevFileFromTypes() ** 
downloading all devfile types from: https://registry.devfile.io 
error: No valid devfile found for project in /home/tpetkos/github/alizer-tests/rhtap-mc-unsuported-runtime/devfile-stack-php-laravel
```

**Comment by thepetk** [comment]
@yangcao77 the workaround has been merged to main and is included in release v1.1.4
